lib/core/test/unit/Cardano/Wallet/DB/SqliteCorruptionSpec.hs

```diff
@@ -0,0 +1,270 @@
+
+    before ((snd <$> fileDBLayer) >>= cleanDB) $
+        describe "Check db reading/writing from/to file and cleaning" $ do
+
+        it "create and list wallet works" $ \db -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            listWallets db `shouldReturn` [testPk]
+            (_, db1) <- fileDBLayer
+            runExceptT (createWallet db1 testPk testCp testMetadata)
+                `shouldReturn` (Left (ErrWalletAlreadyExists testWid))
+            ( testOpeningCleaning
+                listWallets
+                [testPk]
+                [] )
+
+        it "create and get meta works" $ \db -> do
+            now <- getCurrentTime
+            let md = testMetadata { passphraseInfo = Just $ WalletPassphraseInfo now }
+            unsafeRunExceptT $ createWallet db testPk testCp md
+            readWalletMeta db testPk `shouldReturn` Just md
+            ( testOpeningCleaning
+                (`readWalletMeta` testPk)
+                (Just md)
+                Nothing )
+
+        it "create and get private key" $ \db -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            readPrivateKey db testPk `shouldReturn` Nothing
+            let Right phr = fromText "simplephrase"
+                k = unsafeGenerateKeyFromSeed (coerce phr, coerce phr) phr
+            h <- encryptPassphrase phr
+            unsafeRunExceptT (putPrivateKey db testPk (k, h))
+            readPrivateKey db testPk `shouldReturn` Just (k, h)
+            ( testOpeningCleaning
+                (`readPrivateKey` testPk)
+                (Just (k, h))
+                Nothing )
+
+        it "put and read tx history" $ \db -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            runExceptT (putTxHistory db testPk testTxs) `shouldReturn` Right ()
+            readTxHistory db testPk `shouldReturn` testTxs
+            ( testOpeningCleaning
+                (`readTxHistory` testPk)
+                testTxs
+                Map.empty )
+
+        it "put and read checkpoint" $ \db -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            runExceptT (putCheckpoint db testPk testCp) `shouldReturn` Right ()
+            readCheckpoint db testPk `shouldReturn` Just testCp
+            ( testOpeningCleaning
+                (`readCheckpoint` testPk)
+                (Just testCp)
+                Nothing )
+
+    withDB inMemoryDBLayer $
+        describe "random operation chunks property when writing to/reading from file" $ do
+        it "realize a random batch of operations upon one db open"
+            (property . prop_randomOpChunks)
+
+    where
+        testOpeningCleaning
+            :: (Show s, Eq s)
+            => (DBLayer IO (SeqState DummyTarget) DummyTarget -> IO s)
+            -> s
+            -> s
+            -> Expectation
+        testOpeningCleaning call expectedAfterOpen expectedAfterClean = do
```

Comment:
I suspect that cleanDB is already pretty well tested because it's run before every DBSpec test case.

But I think the intention of this is to check that changes made with one connection have an effect on the db file, which can be seen from subsequent connections.

Reply:
cleanDB works fine - I agree, but we had problem with closing DB. See comment below for more details on it